src/unix/tcp.c

```diff
@@ -319,6 +321,15 @@ int uv__tcp_nodelay(int fd, int on) {
 }
 
 
+int uv__tcp_fastopen(int fd, int on) {
+#ifdef TCP_FASTOPEN
+  if (setsockopt(fd, IPPROTO_TCP, TCP_FASTOPEN, &on, sizeof(on)))
+    return -errno;
+#endif
+  return 0;
```

Comment:
this should return ENOTSUP or ENOSYS if it's not supported. The user needs to know.